### DISCUSSION TITLE
How to skip the fill cell step for ASAP7

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#757] [General] [answered]
I know that there's some larger fill cells in the works and that it's not too far away, but meanwhile I want to skip the fill cell for ASAP7 because it greatly incrases the size of the .odb file and slows down the OpenROAD GUI.

There might be a simpler way to do this, but, naively, I came up with the approach below:

```
make results/asap7/foo/base/4_1_cts.odb
cp results/asap7/foo/base/4_1_cts.odb results/asap7/foo/base/4_2_cts_fillcell.odb
make
```

With a bit of rummaging around, I discovered that I can setup a private.mk file:

```
export PRIVATE_DIR = ./designs/$(PLATFORM)/${DESIGN_NICKNAME}
```

Then override the rule that adds fillcells:

```
# Skip fillcell for now.
results/asap7/foo/base/4_2_cts_fillcell.odb: results/asap7/foo/base/4_1_cts.odb
       cp results/asap7/foo/base/4_1_cts.odb results/asap7/foo/base/4_2_cts_fillcell.odb
```

**Comment by vvbandeira** [comment, marked as answer]
@rovinski, do you have any suggestions/opinions here? Would add an `if` statement on the Makefile or the Tcl script is useful to other users, or is the workaround @oharboe using the correct approach?

**Reply by oharboe** [reply]
Perhaps just document, if it isnt already?

It.is of course generic "make" knowledge, but documenting it would make it policy and reduce clutter in the Makefile.

**Reply by rovinski** [reply]
I would simply comment out this line
https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/9895e23b5d4abd4610f8d55ccf8f5173e770375e/flow/scripts/fillcell.tcl#L7

Or for a more reusable approach, change it to:
```
if {![info exists ::env(SKIP_FILL_CELLS)] || $::env(SKIP_FILL_CELLS)} {
  filler_placement $::env(FILL_CELLS)
}
```
and then include `export SKIP_FILL_CELLS = 1` in your design's config.mk.

I don't think it is a good idea to upstream flow support to skip this step. 
1. It is just a hacky workaround and not useful for complete designs
2. It is more of a problem with a non-fabricable PDK than the tools
3. It can lead to other design problems during chip finishing which may confuse users that accidentally leave the flag on.

**Reply by oharboe** [reply]
This is inconvenient as we deploy unmodified docker images from master, ideally to our build system.

The advantage of the private.mk approach is that it works around this problem in a way that doesn't need OpenROAD-flow-scripts to be modified, nor does it pollue the scripts with workaround hooks and hacks.

So I thin things are fine the way they are.

Perhaps the workaround can be here in this forum instead of in the manual. Google will find it :-)
